### PR TITLE
Add prev/next buttons for preset explorer

### PR DESCRIPTION
**Pull request by seri14**
This proposal is reimplement of #100

This request contains
* Add prev/next button
* Change contant of preset path to stem
* Highlight current preset in selectable items ~and can not select it~

preset_explorer_prevnext

### CODE REVIEW

File: source/gui.cpp [hunk 1]
```diff
@@ -2093,40 +2136,39 @@ void reshade::runtime::draw_preset_explorer()
...
 		}
+		for (const auto &entry : preset_files)
+		{
+			const bool is_current_preset_path = entry == _current_preset_path;
+			if (bool selected = is_current_preset_path;  ImGui::Selectable(entry.path().filename().u8string().c_str(), &selected))
+				_file_selection_path = entry, condition = is_current_preset_path ? condition::none : condition::select;
+
+			if (((condition == condition::backward || condition == condition::forward) && entry == _file_selection_path)
+				|| (is_current_preset_path && ImGui::IsWindowAppearing()))
+				ImGui::SetScrollHereY(); // TODO: not work in is_current_preset_path && ImGui::IsWindowAppearing()
```
Comment (seri14) [inline source/gui.cpp]: If use these functions, It requires a window equals to root window.

**Comment by seri14** [comment]
Added changes
* Path area's default state is button. If it press then closing and cancel selection. (To edit path, press button with control key)
* Scroll to current preset after it switching.
* Fixed control focus issues. 
  * after open the preset selector, you can scroll the explorer.
  * after press plus button, you can input name instantly
* Fixed behavior when current preset is not found or selection source is empty.

preset_explorer_prevnext2

**Comment by seri14** [comment]
Demo of use the prev/next buttons
preset_explorer_prevnext_demo

File: source/runtime.cpp [hunk 2]
```diff
@@ -875,7 +885,7 @@ void reshade::runtime::save_config(const std::filesystem::path &path) const
 
 void reshade::runtime::load_preset(const std::filesystem::path &path)
 {
-	const ini_file preset(path);
+	const ini_file preset(g_reshade_dll_path.parent_path() / path);
```
Comment (crosire) [inline source/runtime.cpp]: I'd rather force `_current_preset_path` to always be absolute rather than doing this here.

**Comment by seri14** [comment]
I finished implementation of this feature.
ready for review.

**Pull request merged**